<a href="https://colab.research.google.com/github/K-onkomal/Google-Collab/blob/master/Tf_keras_train_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf

TensorFlow 2.x selected.


Load and prepare the MNIST dataset. Convert the samples from integers to floating-point numbers:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 0s 0us/step


Build the tf.keras.Sequential model by stacking layers. Choose an optimizer and loss function for training:

In [0]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate the model:

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 6s 104us/sample - loss: 0.2931 - accuracy: 0.9156
Epoch 2/5
60000/60000 [==============================] - 5s 89us/sample - loss: 0.1449 - accuracy: 0.9571
Epoch 3/5
60000/60000 [==============================] - 5s 91us/sample - loss: 0.1080 - accuracy: 0.9671
Epoch 4/5
60000/60000 [==============================] - 5s 88us/sample - loss: 0.0882 - accuracy: 0.9722
Epoch 5/5
60000/60000 [==============================] - 5s 87us/sample - loss: 0.0770 - accuracy: 0.9757
10000/1 - 0s - loss: 0.0394 - accuracy: 0.9765


[0.07735454783039168, 0.9765]

The image classifier is now trained to ~98% accuracy on this dataset. To learn more, read the TensorFlow tutorials.